--- wasi.go	2025-02-25 20:08:36
+++ wasi.go	2025-03-10 22:15:41
@@ -112,6 +112,15 @@
 	return errors.New("failed to open file")
 }

+func (c *WasiConfig) SetStdinBytes(data []byte) {
+       bytes := C.wasm_byte_vec_t{
+               size: C.size_t(len(data)),
+               data: (*C.wasm_byte_t)(C.CBytes(data)),
+       }
+       C.wasi_config_set_stdin_bytes(c.ptr(), &bytes)
+       runtime.KeepAlive(c)
+}
+
 func (c *WasiConfig) InheritStdin() {
 	C.wasi_config_inherit_stdin(c.ptr())
 	runtime.KeepAlive(c)
